datadog_lambda/dsm.py

```diff
@@ -1,38 +1,79 @@
+import json
+
+from ddtrace.internal.logger import get_logger
 from datadog_lambda import logger
 from datadog_lambda.trigger import EventTypes
 
+log = get_logger(__name__)
 
-def set_dsm_context(event, event_source):
 
+def set_dsm_context(event, event_source):
     if event_source.equals(EventTypes.SQS):
         _dsm_set_sqs_context(event)
 
 
-def _dsm_set_sqs_context(event):
+def _dsm_set_context_helper(service_type, arn, payload_size, context_json):
+    """
+    Common helper function for setting DSM context.
+
+    Args:
+        service_type: The service type string (example: sqs', 'sns')
+        arn:  ARN from the record
+        payload_size: payload size of the record
+        context_json: Datadog context for the record
+    """
     from datadog_lambda.wrapper import format_err_with_traceback
     from ddtrace.internal.datastreams import data_streams_processor
     from ddtrace.internal.datastreams.processor import DsmPathwayCodec
-    from ddtrace.internal.datastreams.botocore import (
-        get_datastreams_context,
-        calculate_sqs_payload_size,
-    )
+
+    processor = data_streams_processor()
+
+    try:
+        ctx = DsmPathwayCodec.decode(context_json, processor)
+        ctx.set_checkpoint(
+            ["direction:in", f"topic:{arn}", f"type:{service_type}"],
+            payload_size=payload_size,
+        )
+    except Exception as e:
+        logger.error(format_err_with_traceback(e))
```

Comment:
I know this was here previously, but upon more thought, I think we shouldn't use `format_err_with_traceback`. Instead, we should just create a regular string message. The customer doesn't need the full traceback.

```suggestion
        logger.error(f"Unable to set dsm context: {e}")
```